runelite-client/src/test/java/net/runelite/client/plugins/xpglobes/XpGlobesTest.java

```diff
@@ -0,0 +1,135 @@
+package net.runelite.client.plugins.xpglobes;
+
+import com.google.inject.Guice;
+import com.google.inject.testing.fieldbinder.Bind;
+import com.google.inject.testing.fieldbinder.BoundFieldModule;
+import java.util.Collections;
+import javax.inject.Inject;
+import net.runelite.api.Experience;
+import net.runelite.api.Skill;
+import net.runelite.api.events.StatChanged;
+import net.runelite.client.plugins.xptracker.XpTrackerService;
+import net.runelite.client.ui.overlay.OverlayManager;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import static org.mockito.Mockito.when;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class XpGlobesTest
+{
+	@Inject
+	private XpGlobesPlugin xpGlobesPlugin;
+
+	@Mock
+	@Bind
+	private OverlayManager overlayManager;
+
+	@Mock
+	@Bind
+	private XpGlobesOverlay xpGlobesOverlay;
+
+	@Mock
+	@Bind
+	private XpTrackerService xpTrackerService;
+
+	@Mock
+	@Bind
+	private XpGlobesConfig xpGlobesConfig;
+
+	private static final int virtualLevelTotalXp = Experience.getXpForLevel(Experience.MAX_REAL_LEVEL + 1);
+
+	@Before
+	public void before()
+	{
+		Guice.createInjector(BoundFieldModule.of(this)).injectMembers(this);
+
+		statChanged(virtualLevelTotalXp, Skill.AGILITY);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+	}
+
+	@Test
+	public void testVirtualLevelInGlobeIsNotShownByDefault()
+	{
+		when(xpGlobesConfig.showVirtualLevel()).thenReturn(false);
+
+		statChanged(virtualLevelTotalXp + 1, Skill.AGILITY);
+
+		assertEquals(Experience.MAX_REAL_LEVEL, xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testVirtualLevelInGlobeIsShownWhenConfigured()
+	{
+		when(xpGlobesConfig.showVirtualLevel()).thenReturn(true);
+
+		statChanged(virtualLevelTotalXp + 1, Skill.AGILITY);
+
+		assertEquals(Experience.getLevelForXp(virtualLevelTotalXp + 1), xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testGlobeIsNotShownWhenHideMaxAndShowVirtualLevelConfigured()
+	{
+		when(xpGlobesConfig.hideMaxed()).thenReturn(true);
+
+		statChanged(virtualLevelTotalXp + 1, Skill.AGILITY);
+
+		assertEquals(Collections.emptyList(), xpGlobesPlugin.getXpGlobes());
```

Comment:
Use `assertTrue(plugin.getXpGlobes().isEmpty())` for emptiness checks.
```suggestion
		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
```